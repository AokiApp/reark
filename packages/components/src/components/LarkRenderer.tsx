import React, { useContext } from "react";
import { LarkApiContext } from "../contexts/LarkApiContext";
import { BlockStoreProvider } from "../contexts/BlockStoreContext";
import { BlockComponent } from "./BlockComponent";

export const LarkRenderer: React.FC = () => {
  const { blocks } = useContext(LarkApiContext);

  if (!blocks || blocks.length === 0) return null;

  const rootBlock =
    blocks.find((block) => block.block_type === 1 && !block.parent_id) ||
    blocks[0];

  if (!rootBlock) return null;

  return (
    <BlockStoreProvider items={blocks}>
      <BlockComponent blockId={rootBlock.block_id} />
    </BlockStoreProvider>
  );
};
