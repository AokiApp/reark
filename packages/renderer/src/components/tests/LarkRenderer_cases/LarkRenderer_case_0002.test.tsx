import { renderWithVRT } from "../test-utils/renderWithVRT";
import { LarkRenderer } from "../../LarkRenderer";

describe("LarkRenderer 属性バリエーション複合", () => {
  it("textブロック: 太字・斜体・下線・打消し・インラインコード・コメント付き", async () => {
    const textBlock = {
      block_id: "text-attr-1",
      block_type: 2,
      parent_id: "",
      text: {
        elements: [
          {
            text_run: {
              content: "太字",
              text_element_style: { bold: true },
            },
          },
          {
            text_run: {
              content: "斜体",
              text_element_style: { italic: true },
            },
          },
          {
            text_run: {
              content: "下線",
              text_element_style: { underline: true },
            },
          },
          {
            text_run: {
              content: "打消し",
              text_element_style: { strikethrough: true },
            },
          },
          {
            text_run: {
              content: "インラインコード",
              text_element_style: { inline_code: true },
            },
          },
          {
            text_run: {
              content: "コメント付き",
              text_element_style: { comment_ids: ["c1"] },
            },
          },
        ],
        style: { align: 1, folded: false },
      },
      children: [],
    };

    const blocks = [textBlock];

    const { container, vrt } = renderWithVRT(
      <LarkRenderer initialData={{ blocks }} />,
    );
    await vrt();
    expect(container).toMatchSnapshot();
  });
});
