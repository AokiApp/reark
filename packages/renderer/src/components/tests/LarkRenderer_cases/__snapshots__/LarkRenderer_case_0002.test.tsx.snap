// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LarkRenderer 属性バリエーション複合 textブロック: 太字・斜体・下線・打消し・インラインコード・コメント付き 1`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <span
        class="reark-text--bold reark-text"
      >
        太字
      </span>
      <span
        class="reark-text--italic reark-text"
      >
        斜体
      </span>
      <span
        class="reark-text--underline reark-text"
      >
        下線
      </span>
      <span
        class="reark-text--strikethrough reark-text"
      >
        打消し
      </span>
      <span
        class="reark-text--inline-code reark-text"
      >
        インラインコード
      </span>
      <div
        class="reark-comment"
        data-comment-ids="c1"
      >
        <span
          class="reark-text"
        >
          コメント付き
        </span>
      </div>
    </div>
  </div>
</div>
`;
